WIP create sign in page

jdavidrice opened the pull request:
## Proposed changes

I created a basic sign in page that isn't finished completely. We will need to add the navbar and footer. 

## PR Checklist

<!-- ℹ️ Put an `X` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. ℹ️ -->

- [x] I have read the [CONTRIBUTING](https://github.com/davidsaulrodriguez/shop-portal/blob/master/CONTRIBUTING.md) doc
- [x] I have read the [BRANCHING MODEL](https://github.com/davidsaulrodriguez/shop-portal/blob/master/BRANCHINGMODEL.md) doc
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## PR Type

<!-- Please check the one (or all) that applies to this PR using "x". -->

What types of changes does your code introduce to **Shop Portal**?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code Style Update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (Please describe):

## Current behavior

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #11 

## New behavior

<!-- Please describe the new behavior that you are introducing, or link to a relevant issue. -->

Fixes N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

jdavidrice committed:
Add test button to app.js

jdavidrice committed:
WIP attempt to create login page

jdavidrice committed:
Remove unneeded elements from SignIn page

jdavidrice committed:
Remove copyright and add Facebook and Google labels for future sign in options

jdavidrice committed:
Add placeholder buttons for Facebook and Google and adjust vertical spacing

jdavidrice committed:
Merge branch 'main' into signupLoginPage

jdavidrice committed:
Add tests folder and added path to test script

jdavidrice committed:
Add SignIn test file

jdavidrice committed:
Change path for test script in client/package.json

jdavidrice committed:
Remove unneeded semicolons from test files

davidsaulrodriguez commented:
@jdavidrice - Where are all of these packages coming from? What do they do?

```
"base-shell": "^1.2.21",
"jss-rtl": "^0.3.0",
"material-ui-shell": "^1.3.2",
"react-custom-scrollbars": "^4.2.1",
"react-intl": "^5.15.6",
"react-router-dom": "^5.2.0",
"react-virtualized-auto-sizer": "^1.0.5",
"react-window": "^1.8.6"
```

I see that everything is dark-themed but nothing in the styles indicate that a dark theme should be used. I'm confused as to how all of this is happening. Could I get some more info on all of this and the decisions made.

**Screenshot of what I see when starting this branch**
![Screenshot_20210404_115155](https://user-images.githubusercontent.com/4968000/113514393-384bcc00-953c-11eb-96b0-03ac2831faf1.png)

jdavidrice commented:
The packages that you listed were left over from an earlier version of the branch. I removed them from the client package.json but did not realize that they were also listed in the server package.json. I will remove them. 

The dark background color on the page was due to the background-color property for the div containing the SignIn component in the App.css file. I changed it to white and removed some other unneeded styles from that file. I also made the h1 elements black so they show up on the white background.

jdavidrice committed:
Remove unneeded packages and change SignIn to white background

jdavidrice committed:
Change color of Remember Me text on SignIn page to black

jdavidrice committed:
Remove unneeded color from FormControlLabel in SignIn.jsx

jdavidrice committed:
Add in Navigation and NavigationDrawer components

jdavidrice committed:
Revert "Remove unneeded color from FormControlLabel in SignIn.jsx"
Error: Invariant failed: You should not use <withRouter(Navigation) /> outside a <Router>
This reverts commit ad633c0db8f8b82dc393c3706f9b50cfb34b4421.

jdavidrice commented:
This PR was still WIP so I'm surprised it was merged. I will still be working on this feature today and will create another PR later.
